modules/workflow-workflowoperation/src/test/java/org/opencastproject/workflow/handler/workflow/ConditionalConfigWorkflowOperationHandlerTest.java

```diff
@@ -0,0 +1,124 @@
+    workflowInstance.setMediaPackage(MediaPackageBuilderFactory.newInstance().newMediaPackageBuilder().createNew());
+
+    operation = new WorkflowOperationInstanceImpl("conditional-config", OperationState.RUNNING);
+    operation.setConfiguration(ConditionalConfigWorkflowOperationHandler.CONFIGURATION_NAME, CONFIGURATION_VAR);
+    operation.setConfiguration(ConditionalConfigWorkflowOperationHandler.CONDITION_PREFIX + "-1", "${step} == 1");
+    operation.setConfiguration(ConditionalConfigWorkflowOperationHandler.VALUE_PREFIX + "-1", "value 1");
+    operation.setConfiguration(ConditionalConfigWorkflowOperationHandler.CONDITION_PREFIX + "-2", "${step} == 2");
+    operation.setConfiguration(ConditionalConfigWorkflowOperationHandler.VALUE_PREFIX + "-2", "value 2");
+    operation.setConfiguration(ConditionalConfigWorkflowOperationHandler.CONDITION_PREFIX + "-3", "${step} == 3");
+    operation.setConfiguration(ConditionalConfigWorkflowOperationHandler.VALUE_PREFIX + "-3", "value 3");
+    operation.setConfiguration(ConditionalConfigWorkflowOperationHandler.NO_MATCH, "value no-match");
+
+    List<WorkflowOperationInstance> operationsList = new ArrayList<WorkflowOperationInstance>();
+    operationsList.add(operation);
+    workflowInstance.setOperations(operationsList);
+  }
+
+  @After
+  public void tearDown() throws Exception {
+  }
+
+  @Test
+  public void testFirstConditionTrue() throws Exception {
+    workflowInstance = replaceVars(workflowInstance, "step", "1");
+
+    WorkflowOperationResult result = operationHandler.start(workflowInstance, null);
+    Assert.assertEquals(Action.CONTINUE, result.getAction());
+
+    Assert.assertEquals("value 1", result.getProperties().get(CONFIGURATION_VAR));
+  }
+
+  @Test
+  public void testThirdConditionTrue() throws Exception {
+    workflowInstance = replaceVars(workflowInstance, "step", "3");
+
+    WorkflowOperationResult result = operationHandler.start(workflowInstance, null);
+    Assert.assertEquals(Action.CONTINUE, result.getAction());
+
+    Assert.assertEquals("value 3", result.getProperties().get(CONFIGURATION_VAR));
+  }
+
+  @Test
+  public void testNoConditionTrue() throws Exception {
+    workflowInstance = replaceVars(workflowInstance, "step", "99");
+
+    WorkflowOperationResult result = operationHandler.start(workflowInstance, null);
+    Assert.assertEquals(Action.CONTINUE, result.getAction());
+
+    Assert.assertEquals("value no-match", result.getProperties().get(CONFIGURATION_VAR));
+  }
+
+  private WorkflowInstance replaceVars(WorkflowInstance wf, String key, String value) throws WorkflowParsingException {
+    HashMap<String, String> wfConfig = new HashMap<String, String>();
+    wfConfig.put(key, value);
+
+    // Replace operation configuration with variables form wf config
+    final Function<String, String> systemVariableGetter = k -> null;
+    String xml = WorkflowConditionInterpreter.replaceVariables(WorkflowParser.toXml(workflowInstance),
```

Comment:
`workflowInstance` is used directly.
```suggestion
    String xml = WorkflowConditionInterpreter.replaceVariables(WorkflowParser.toXml(wf),
```

Reply:
Good catch!